src/utils/std.rs

```diff
@@ -9,3 +10,16 @@ use crate::lib::String;
 pub trait StellarTypeToString<T, E: From<sp_std::str::Utf8Error>> {
     fn as_encoded_string(&self) -> Result<String, E>;
 }
+
+/// A trait used to convert a structure into Base64 String
+pub trait StellarTypeToBase64String {
+    /// returns a Base64 encoded String or a vec of bytes [xdr]
+    fn as_base64_encoded_string(&self) -> String;
+}
+
+impl <T: XdrCodec> StellarTypeToBase64String for T {
+    fn as_base64_encoded_string(&self) -> String {
+        let xdr = self.to_base64_xdr();
+        String::from_utf8(xdr.clone()).unwrap_or(format!("{:?}", xdr))
```

Comment:
> Does the or case ever happen?

True, it will not. 

I have reverted std.rs file and use `unwrap()`.
I do not think it is bad practice, as long as explained well.